nilearn/regions/connectivity.py

```diff
@@ -0,0 +1,392 @@
+    vndx = np.nonzero(np.var(func_data, 1) != 0)[0]
+    iv = iv[vndx]
+    m = len(iv)
+    print(m, " # of non-zero valued or non-zero variance voxels in the mask")
+
+    # Construct a sparse matrix from the mask
+    msk = csc_matrix(
+        (vndx + 1, (iv, np.zeros(m))), shape=(prod(msz), 1), dtype=np.float32
+    )
+
+    sparse_i = []
+    sparse_j = []
+    sparse_w = [[]]
+
+    for i in range(0, m):
+        if i % 1000 == 0:
+            print("voxel #", i)
+
+        # Convert index into 3D and calculate neighbors, then convert resulting
+        # 3D indices into 1D
+        ndx1d = indx_3dto1d(indx_1dto3d(iv[i], sz[:-1]) + neighbors, sz[:-1])
+
+        # Convert 1D indices into masked versions
+        ondx1d = msk[ndx1d].todense()
+
+        # Exclude indices not in the mask
+        ndx1d = ndx1d[np.nonzero(ondx1d)[0]].flatten()
+        ondx1d = np.array(ondx1d[np.nonzero(ondx1d)[0]])
+        ondx1d = ondx1d.flatten() - 1
+
+        # Keep track of the index corresponding to the "seed"
+        nndx = np.nonzero(ndx1d == iv[i])[0]
+
+        # Extract the time courses corresponding to the "seed" and 3D
+        # neighborhood voxels
+        tc = np.array(func_data[ondx1d.astype("int"), :])
+
+        # Ensure that the "seed" has variance, if not just skip it
+        if np.var(tc[nndx, :]) == 0:
+            continue
+
+        # Calculate functional connectivity maps for "seed" and 3D neighborhood
+        # voxels
+        R = np.corrcoef(np.dot(tc, func_data.T) / (sz[3] - 1))
+
+        if np.linalg.matrix_rank(R) == 1:
+            R = np.reshape(R, (1, 1))
+
+        # Set nans to 0
+        R[np.isnan(R)] = 0
+
+        # Set values below thresh to 0
+        R[R < thresh] = 0
+
+        # Calculate the spatial correlation between FC maps
+        if np.linalg.matrix_rank(R) == 0:
+            R = np.reshape(R, (1, 1))
+
+        # Keep track of the indices and the correlation weights to construct
+        # sparse connectivity matrix
+        sparse_i = np.append(sparse_i, ondx1d, 0)
+        sparse_j = np.append(sparse_j, (ondx1d[nndx]) * np.ones(len(ondx1d)))
+        sparse_w = np.append(sparse_w, R[nndx, :], 1)
+
+    # Ensure that the weight vector is the correct shape
+    sparse_w = np.reshape(sparse_w, prod(np.shape(sparse_w)))
+
+    # Concatenate the i, j, and w_ij vectors
+    outlist = sparse_i
+    outlist = np.append(outlist, sparse_j)
+    outlist = np.append(outlist, sparse_w)
+
+    # Calculate the number of non-zero weights in the connectivity matrix
+    n = len(outlist) / 3
+
+    # Reshape the 1D vector read in from infile in to a 3xN array
+    outlist = np.reshape(outlist, (3, int(n)))
+
+    m = max(max(outlist[0, :]), max(outlist[1, :])) + 1
+
+    W = csc_matrix(
+        (outlist[2, :], (outlist[0, :], outlist[1, :])),
+        shape=(int(m), int(m)),
+        dtype=np.float32,
+    )
+
+    return W
+
+
+def make_local_connectivity_tcorr(func_img, clust_mask_img, thresh):
+    """
```

Comment:
All the comments apply to this function. what is actually the dfference btx _scorr and _tcorr ?

Reply:
T-Corr permits defining connectivity on the basis of voxel temporal correlation and S-corr permits defining connectivity on the basis of voxel spatial correlation.